grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -1624,6 +1624,9 @@ class StockMovementService {
         shipment.destination = stockMovement.destination
         shipment.shipmentType = ShipmentType.get(Constants.DEFAULT_SHIPMENT_TYPE_ID)
 
+        // Save shipment before adding the items to avoid dead locks
+        shipment.save()
```

Comment:
Why would there be deadlocks here? Are you actually seeing deadlocks?

Reply:
Actually, it was not a `dead lock` issue, but `referencing an unsaved transient instance` and I simply moved this save from the 1648 line to be done once, instead of for each item. I'll fix the comment.